pytorch_toolkit/action_recognition/utils/preprocess_videos.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import division, print_function
+
+import json
+import os
+import subprocess
+from argparse import ArgumentParser
+from functools import partial
+from multiprocessing import Pool
+from pathlib import Path
+
+import cv2
+from tqdm import tqdm
+
+FFMPEG_CMD = "ffmpeg -i \"{input!s}\" -q:v {q} {filters} -an -threads 1 -y -loglevel panic \"{output!s}\""
+SCALE_OPT = "-vf \"scale=trunc(((iw*{scale})/'min(iw,ih)')/2)*2:trunc(((ih*{scale})/'min(iw,ih)')/2)*2\""
+
+
+def get_video_properties(video_path):
+    cap = cv2.VideoCapture(video_path)
+    if not cap.isOpened():
+        # video file exists but contains invalid video stream
+        return None, None
+
+    n_frames = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+    fps = cap.get(cv2.CAP_PROP_FPS)
+    return n_frames, fps
+
+
+def convert_video(video, output, scale, quality):
+    if scale:
+        filters_opt = SCALE_OPT.format(scale=scale)
+    else:
+        filters_opt = ""
+
+    cmd_format = FFMPEG_CMD.format(input=video, output=output, filters=filters_opt, q=quality)
+    subprocess.call(cmd_format, shell=True)
+
+
+def callback(data, orig_path, destination_path, scale=None, quality=1, video_format='frames', convert=True):
+    video_name, annotation = data
+
+    if annotation['subset'] == 'testing':
+        label = 'test'
+    else:
+        label = annotation['annotations']['label']
+
+    class_path = os.path.join(destination_path, label)
+    Path(class_path).mkdir(exist_ok=True)
+
+    orig_video_path = os.path.join(orig_path, label, video_name) + ".mp4"
```

Comment:
It's not work for  `UCF-101` dataset that contains video with `.avi` extension

Reply:
thanks, fixed